controllers/consoleplugin/consoleplugin_reconciler.go

```diff
@@ -81,49 +81,93 @@ func (r *CPReconciler) Reconcile(ctx context.Context, desired *flowslatest.FlowC
 		return err
 	}
 
-	// Create object builder
-	builder := newBuilder(ns, r.Instance.Image, &desired.Spec)
+	if helper.UseConsolePlugin(&desired.Spec) {
+		// Create object builder
+		builder := newBuilder(ns, r.Instance.Image, &desired.Spec)
 
-	if err := r.reconcilePermissions(ctx, &builder); err != nil {
-		return err
-	}
+		if err := r.reconcilePermissions(ctx, &builder); err != nil {
+			return err
+		}
 
-	if err = r.reconcilePlugin(ctx, &builder, &desired.Spec); err != nil {
-		return err
-	}
+		if err = r.reconcilePlugin(ctx, &builder, &desired.Spec); err != nil {
+			return err
+		}
 
-	cmDigest, err := r.reconcileConfigMap(ctx, &builder, &desired.Spec)
-	if err != nil {
-		return err
-	}
+		cmDigest, err := r.reconcileConfigMap(ctx, &builder, &desired.Spec)
+		if err != nil {
+			return err
+		}
 
-	if err = r.reconcileDeployment(ctx, &builder, &desired.Spec, cmDigest); err != nil {
-		return err
-	}
+		if err = r.reconcileDeployment(ctx, &builder, &desired.Spec, cmDigest); err != nil {
+			return err
+		}
 
-	if err = r.reconcileServices(ctx, &builder, &desired.Spec); err != nil {
-		return err
-	}
+		if err = r.reconcileServices(ctx, &builder, &desired.Spec); err != nil {
+			return err
+		}
 
-	if err = r.reconcileHPA(ctx, &builder, &desired.Spec); err != nil {
-		return err
+		if err = r.reconcileHPA(ctx, &builder, &desired.Spec); err != nil {
+			return err
+		}
+
+		// Watch for Loki certificates if necessary; we'll ignore in that case the returned digest, as we don't need to restart pods on cert rotation
+		// because certificate is always reloaded from file
+		if _, err = r.Watcher.ProcessCACert(ctx, r.Client, &desired.Spec.Loki.TLS, r.Namespace); err != nil {
+			return err
+		}
+		if _, _, err = r.Watcher.ProcessMTLSCerts(ctx, r.Client, &desired.Spec.Loki.StatusTLS, r.Namespace); err != nil {
+			return err
+		}
+	} else {
+		// delete any existing owned object
+		return r.deleteOwned(ctx)
 	}
 
-	// Watch for Loki certificates if necessary; we'll ignore in that case the returned digest, as we don't need to restart pods on cert rotation
-	// because certificate is always reloaded from file
-	if _, err = r.Watcher.ProcessCACert(ctx, r.Client, &desired.Spec.Loki.TLS, r.Namespace); err != nil {
-		return err
+	return nil
+}
+
+func (r *CPReconciler) deleteOwned(ctx context.Context) error {
```

Comment:
there should be `r.Managed.TryDeleteAll(ctx)` that does pretty much the same thing already, I think (it's used in FLP reconcilers)

Reply:
oh, thanks for sharing that !

Reply:
Done in https://github.com/netobserv/network-observability-operator/pull/354/commits/36eb0f243f6e43633a3731d7a313fb5feb1a1b69